src/nvim/CMakeLists.txt

```diff
@@ -234,6 +234,11 @@ list(APPEND NVIM_LINK_LIBRARIES
     ${CMAKE_THREAD_LIBS_INIT}
     )
 
+if(WIN32)
+  # libuv needs these
+  list(APPEND NVIM_LINK_LIBRARIES iphlpapi psapi userenv ws2_32)
+endif(WIN32)
```

Comment:
Anyone feel free to correct me, but I think the place for this is in `cmake/FindLibUV.cmake`. It defines `LIBUV_LIBRARIES` (which is later used to build `NVIM_LINK_LIBRARIES` on line 225, above). Another nice thing about `FindLibUV.cmake` is that it calls `check_library_exists()` and then defines appropriate `HAVE_LIB` symbols.

If possible we should continue following the example @jszakmeister set.